subsys/net/lib/nrf_cloud/src/nrf_cloud_agnss.c

```diff
@@ -465,6 +465,13 @@ static size_t get_next_agnss_element(struct nrf_cloud_agnss_element *element,
 	 * each element.
 	 */
 	if (elements_left_to_process == 0) {
+		/* If there's no more data, reset the parser state. */
+		if (buf_len == 0) {
+			elements_left_to_process = 0;
+			element_type = NRF_CLOUD_AGNSS__TYPE_INVALID;
+			return 0;
+		}
```

Comment:
I don't fully follow the operation of the AGNSS/PGNSS parsers, but:

It seems like 

```c
if (buf_len < NRF_CLOUD_AGNSS_BIN_TYPE_SIZE + NRF_CLOUD_AGNSS_BIN_COUNT_SIZE) {
			LOG_ERR("Unexpected end of data");
			return 0;
}
```

Also detects this?

Would placing `element_type = NRF_CLOUD_AGNSS__TYPE_INVALID;` inside that branch accomplish the same effect?

Reply:
I think it would be good to also set `element_type` to invalid in the unexpected end of data case, I missed that one.

These two if's are for different cases. The first one is for the success case, where all data was parsed and there's no data left. The second one is for the case where there's still data left, but not enough for the type and count fields. In both cases the function returns 0 to indicate that no next element was found, but in the latter case there should be an error trace.

`element_type` can be set before the if's and `elements_left_to_process` is already zero, so no need to set it. I'll update the implementation.